src/PIL/ImageGrab.py

```diff
@@ -43,17 +43,27 @@ def grab(
             fh, filepath = tempfile.mkstemp(".png")
             os.close(fh)
             args = ["screencapture"]
-            if bbox:
+            if window:
+                args += ["-l", str(window)]
+            # -R is not working with -l
+            if bbox and not window:
                 left, top, right, bottom = bbox
                 args += ["-R", f"{left},{top},{right-left},{bottom-top}"]
             subprocess.call(args + ["-x", filepath])
             im = Image.open(filepath)
             im.load()
             os.unlink(filepath)
             if bbox:
-                im_resized = im.resize((right - left, bottom - top))
-                im.close()
-                return im_resized
+                # manual crop for windowed mode
```

Comment:
There's an interesting problem here.

Originally, we did just crop the image using Pillow. However, that changed in #6152, because #6144 reported that it wasn't accounting for scaling on retina displays.

So if you pick a specific window, how would we know whether it was scaled or not? It might be on a retina display or it might not. If this were Swift, I'd say we want the [`backingScaleFactor`](https://developer.apple.com/documentation/appkit/nswindow/backingscalefactor) for the window. I'm not convinced AppleScript could easily get information about a window just from the id.

Reply:
I created https://github.com/yankeguo/Pillow/pull/2

One of the `screencapture` flags is
> -o      In window capture mode, do not capture the shadow of the window.

Excluding the shadow changes the size of the window. So if we take two screenshots of a window, we can measure the size of the shadow.

On a retina screen, the shadow is twice as wide as it is on a non-retina screen.

It is possible that a window doesn't have a shadow, but perhaps that can be considered an edge case.